internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -240,29 +255,20 @@ func WithRecorder(er event.Recorder) ReconcilerOption {
 	}
 }
 
-// A PollIntervalHook determines how frequently the XR should poll its composed
-// resources.
-type PollIntervalHook func(ctx context.Context, xr *composite.Unstructured) time.Duration
-
-// WithPollIntervalHook specifies how to determine how long the Reconciler
-// should wait before queueing a new reconciliation after a successful
-// reconcile.
-func WithPollIntervalHook(h PollIntervalHook) ReconcilerOption {
+// WithFeatures specifies what feature flags the Reconciler should enable.
+func WithFeatures(f *feature.Flags) ReconcilerOption {
 	return func(r *Reconciler) {
-		r.pollInterval = h
+		r.features = f
 	}
 }
 
 // WithPollInterval specifies how long the Reconciler should wait before
 // queueing a new reconciliation after a successful reconcile. The Reconciler
-// uses the interval jittered +/- 10% when all composed resources are ready. It
-// polls twice as frequently (i.e. at half the supplied interval) +/- 10% when
-// waiting for composed resources to become ready.
+// uses the interval jittered +/- 10%.
 func WithPollInterval(interval time.Duration) ReconcilerOption {
-	return WithPollIntervalHook(func(_ context.Context, _ *composite.Unstructured) time.Duration {
-		// Jitter the poll interval +/- 10%.
-		return interval + time.Duration((rand.Float64()-0.5)*2*(float64(interval)*0.1)) //nolint:gosec // No need for secure randomness
```

Comment:
Wouldn't this still be needed if someone needs/decides to disable the beta feature for whatever reason? I am particularly concerned about resource consumption/scalability of real time compositions and wondering if there is a way to get back to the previous configuration if something goes wrong.

Reply:
The definition reconciler still sets a poll interval when realtime compositions is disabled. The jittering happens inline - I wanted to remote `pollIntervalHook` as a simplification.